File: videojuegoDeChinpokomon/src/chimpokomon/GarraMecanica.java
package chimpokomon;

public class GarraMecanica extends Ataque {
    // Esta mal pasarle el chimpokomonAtacante aca
    // Nunca se usa el chimpokomonAtacante
    @Override
    public void dañar(Chimpokomon chinpokomon, Chimpokomon chimpokomonAtacante) {
        double random = Math.random();

        if (chinpokomon.getNaturaleza() == "Animal" && chimpokomonAtacante.getNaturaleza() == "Robot") {
            if (random < 0.5) {
                chinpokomon.restarVida((int) (chinpokomon.nivelDeVida * 0.5) + 2);
                System.out.println(chimpokomonAtacante.nombre + " Realizo un ataque critico");
            } else {
                chinpokomon.restarVida(4);
            }

        }
        if (chinpokomon.getNaturaleza() == "Cosa" && chimpokomonAtacante.getNaturaleza() == "Animal") {
            if (random < 0.5) {
                chinpokomon.restarVida((int) (chinpokomon.nivelDeVida * 0.5) + 2);
                System.out.println(chimpokomonAtacante.nombre + " Realizo un ataque critico");
            } else {
                chinpokomon.restarVida(4);
            }
        }
        if (chinpokomon.getNaturaleza() == "Robot" && chimpokomonAtacante.getNaturaleza() == "Cosa") {
            if (random < 0.5) {
                chinpokomon.restarVida((int) (chinpokomon.nivelDeVida * 0.5) + 2);
                System.out.println(chimpokomonAtacante.nombre + " Realizo un ataque critico");
            } else {
                chinpokomon.restarVida(4);
            }
        }

        else {
            chinpokomon.restarVida(2);
        }

    }
}

File: ChimpokomonPython/ModuloSRC/GarraMecanica.py
from Ataque import Ataque
from Chimpokomon import Chimpokomon
import random
import Ataque as Ataque
import chimpokomon as Chimpokomon


class GarraMecanica(Ataque):
    def dañar(self,  chinpokomon: Chimpokomon, chimpokomonAtacante: Chimpokomon):
        ram = random.randint(0, 1)
        if chinpokomon.getNaturaleza() == "Animal" and chimpokomonAtacante.getNaturaleza() == "Robot" or chinpokomon.getNaturaleza() == "Cosa" and chimpokomonAtacante.getNaturaleza() == "Animal" or chinpokomon.getNaturaleza() == "Robot" and chimpokomonAtacante.getNaturaleza() == "Cosa":
            if ram < 0.5:
                chinpokomon.restarVida(
                    (chinpokomon.getNivelDeVida() * 0.5) + 2)
                print(chimpokomonAtacante.getNombre() +
                      " Realizo un ataque critico")
            else:
                chinpokomon.restarVida(4)
        else:
            chinpokomon.restarVida(2)
